core/merge.js

```diff
@@ -349,6 +360,9 @@ class Merge {
 
       let existingDstRev = existingDstRevs[dst._id]
       if (existingDstRev && folder.overwrite) dst._rev = existingDstRev
+      if (side === 'remote' && newRevs && newRevs[dst.remote._id]) {
+        dst.remote._rev = newRevs[dst.remote._id]
```

Comment:
Would it be possible for `dst.remote` to be null in case a child was added locally but not yet synced remotely?
Wouldn't using lodash's `get` work when `newRevs` and/or `dst.remote` are null?

```js
const newRemoteRev = _.get(newRevs, _.get(dst, 'remote._id'))
if (newRemoteRev) dst.remote._rev = newRemoteRev
```